packages/venia-ui/lib/components/MiniCart/ProductList/productList.js

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo } from 'react';
+import { string, func, arrayOf, shape, bool, number } from 'prop-types';
+
+import LoadingIndicator from '../../LoadingIndicator';
+import Item from './item';
+import { mergeClasses } from '../../../classify';
+
+import defaultClasses from './productList.css';
+
+const ProductList = props => {
+    const { items, loading, handleRemoveItem, classes: propClasses } = props;
+    const classes = mergeClasses(defaultClasses, propClasses);
+
+    const cartItems = useMemo(() => {
+        if (!loading && items) {
+            return items.map(item => (
+                <Item
+                    key={item.id}
+                    {...item}
+                    handleRemoveItem={handleRemoveItem}
+                />
+            ));
+        }
+    }, [items, loading, handleRemoveItem]);
+
+    if (loading) {
+        return <LoadingIndicator>{`Fetching Items in Cart`}</LoadingIndicator>;
+    }
+
+    return <div className={classes.root}>{cartItems}</div>;
```

Comment:
Did we ever land on a desired UX for empty cart? The current looks a bit strange as there's a box with a scroll bar and nothing in it.

[![Image from Gyazo](https://i.gyazo.com/bb0f2c8c846e6b71437bb995c490ffd4.png)](https://gyazo.com/bb0f2c8c846e6b71437bb995c490ffd4)

Reply:
Yes, I believe we wanted a simple message in there that says "There are no items in your cart"?
Ignore the height of the cart in the mock, and just use the min-height we have determined.

![Desktop - Mini Cart - No items](https://user-images.githubusercontent.com/25960021/86649945-d3ad1200-bfa7-11ea-85f1-a0c0212691c3.png)

Reply:
This has been handled in #2511 already.